pkg/operator/ceph/object/user/controller.go

```diff
@@ -0,0 +1,229 @@
+*/
+
+// Package objectuser to manage a rook object store user.
+package objectuser
+
+import (
+	"fmt"
+	"reflect"
+
+	"github.com/coreos/pkg/capnslog"
+	opkit "github.com/rook/operator-kit"
+	cephv1beta1 "github.com/rook/rook/pkg/apis/ceph.rook.io/v1beta1"
+	"github.com/rook/rook/pkg/clusterd"
+	cephrgw "github.com/rook/rook/pkg/daemon/ceph/rgw"
+	"github.com/rook/rook/pkg/operator/k8sutil"
+	"k8s.io/api/core/v1"
+	apiextensionsv1beta1 "k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/v1beta1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/tools/cache"
+)
+
+const (
+	customResourceName       = "objectstoreuser"
+	customResourceNamePlural = "objectstoreusers"
+)
+
+var logger = capnslog.NewPackageLogger("github.com/rook/rook", "op-object")
+
+// ObjectStoreResource represents the object store user custom resource
+var ObjectStoreUserResource = opkit.CustomResource{
+	Name:    customResourceName,
+	Plural:  customResourceNamePlural,
+	Group:   cephv1beta1.CustomResourceGroup,
+	Version: cephv1beta1.Version,
+	Scope:   apiextensionsv1beta1.NamespaceScoped,
+	Kind:    reflect.TypeOf(cephv1beta1.ObjectStoreUser{}).Name(),
+}
+
+// ObjectStoreUserController represents a controller object for object store user custom resources
+type ObjectStoreUserController struct {
+	context     *clusterd.Context
+	rookImage   string
+	hostNetwork bool
+	ownerRef    metav1.OwnerReference
+}
+
+// NewObjectStoreUserController create controller for watching object store user custom resources created
+func NewObjectStoreUserController(context *clusterd.Context, rookImage string, hostNetwork bool, ownerRef metav1.OwnerReference) *ObjectStoreUserController {
+	return &ObjectStoreUserController{
+		context:     context,
+		rookImage:   rookImage,
+		hostNetwork: hostNetwork,
+		ownerRef:    ownerRef,
+	}
+}
+
+// StartWatch watches for instances of ObjectStoreUser custom resources and acts on them
+func (c *ObjectStoreUserController) StartWatch(namespace string, stopCh chan struct{}) error {
+
+	resourceHandlerFuncs := cache.ResourceEventHandlerFuncs{
+		AddFunc: c.onAdd,
+		// UpdateFunc: c.onUpdate,
+		DeleteFunc: c.onDelete,
+	}
+
+	logger.Infof("start watching object store user resources in namespace %s", namespace)
+	watcher := opkit.NewWatcher(ObjectStoreUserResource, namespace, resourceHandlerFuncs, c.context.RookClientset.CephV1beta1().RESTClient())
+	go watcher.Watch(&cephv1beta1.ObjectStoreUser{}, stopCh)
+
+	return nil
+}
+
+func (c *ObjectStoreUserController) onAdd(obj interface{}) {
+	user, err := getObjectStoreUserObject(obj)
+	if err != nil {
+		logger.Errorf("failed to get objectstoreuser object: %+v", err)
+		return
+	}
+
+	if err = c.createUser(c.context, user); err != nil {
+		logger.Errorf("failed to create object store user %s. %+v", user.Name, err)
+	}
+}
+
+// TODO: Need to think through how to do updates, leaving out for now.
+
+// func (c *ObjectStoreUserController) onUpdate(oldObj, newObj interface{}) {
```

Comment:
Makes sense to not worry about implementing the update function until more features are added where it would be useful. For now, I'd say our general pattern would be to uncomment the function and leave it empty (or with a comment), but not commit all the commented lines of code.